debian/postrm

```diff
@@ -19,7 +19,6 @@ remove_logs(){
 remove_esm(){
     # config files created in postinst, need explicit handling on purge
     rm -f /etc/apt/trusted.gpg.d/ubuntu-advantage-esm-infra-trusty.gpg
-    rm -f /etc/apt/sources.list.d/ubuntu-esm-infra-trusty.list
```

Comment:
So this remove_esm was trusty only in nature because it because historically it was defined that a customer would have to ua attach again on the new Ubuntu Release, plus the gpg keys used in trusty repos were to be different than those used in xenial+. Those assumptions have changed a little as we now have to handle upgrade path from trusty -> focal and remain attached.

What I think this means is that these purge functions need to be flexible enough to handle any apt config and prefs for any ua client service on any series. So 

- [ ] I think we need to generalize these file matches a bit to remove apt config for any entitled UA service upon purge. 

- [ ] Also I think we need to avoid encoding the ubuntu series in apt config files https://github.com/canonical/ubuntu-advantage-client/blob/master/uaclient/entitlements/repo.py#L34-L45.

- [ ] We need to adapt debian/postinst to rename apt trusty|xenial|bionic|focal.list and preference files one more time to remove the encoded <release> name from the apt config files.